File: src/test/java/name/seguri/java/leetcode/_125/SolutionTest.java
package name.seguri.java.leetcode._125;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class SolutionTest {

  private final Solution sut = new Solution();

  @Test
  void input1() {
    var result = sut.isPalindrome("A man, a plan, a canal: Panama");

    assertThat(result).isTrue();
  }

  @Test
  void input2() {
    var result = sut.isPalindrome("race a car");

    assertThat(result).isFalse();
  }

  @Test
  void input3() {
    var result = sut.isPalindrome(" ");

    assertThat(result).isTrue();
  }
}

File: src/test/kotlin/name/seguri/kotlin/leetcode/_125/SolutionTest.kt
package name.seguri.kotlin.leetcode._125

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SolutionTest {
  private val sut = Solution()

  @Test
  fun input1() {
    val result = sut.isPalindrome("A man, a plan, a canal: Panama")
    assertThat(result).isTrue()
  }

  @Test
  fun input2() {
    val result = sut.isPalindrome("race a car")
    assertThat(result).isFalse()
  }

  @Test
  fun input3() {
    val result = sut.isPalindrome(" ")
    assertThat(result).isTrue()
  }
}
